src/Home/Home.tsx

```diff
@@ -1,20 +1,25 @@
 import React from "react";
 import { Link, RouteComponentProps } from "react-router-dom";
-
-import { IParams } from "src/common/interface";
+import Button from "src/components/Button";
+import { useAuth } from "src/context/GlobalContext";
 
 interface IPage {
-  name: string
+  name: string;
 }
 
-const HomePage: React.FunctionComponent<IPage & RouteComponentProps<IParams>> =
-  ({ name }) => {
-    return (
-      <div>
-        <p>{name}</p>
-        <Link to="/about">Go to the About page!</Link>
-      </div>
-    );
-  };
+const HomePage: React.FunctionComponent<IPage & RouteComponentProps> = ({
+  name,
+}) => {
+  const { state, actions } = useAuth();
+
+  return (
+    <div>
+      <p>{name}</p>
+      <p>Welcome {state.user?.getProfile().email}</p>
```

Comment:
`user` is a required dependency

Reply:
Initially we wont be having user in our context so the type use user is `User | null` that's why the editor automatically detects it while using to avoid null case.